server/pull_request.go

```diff
@@ -457,6 +469,44 @@ func (s *Server) CleanOutdatedPRs() {
 	mlog.Info("Finished update the outdated prs in the mattermod database....")
 }
 
+func (s *Server) CleanUpLabels(client *GithubClient, pr *model.PullRequest) {
```

Comment:
I thought about returning an `error`, but that would mean that the caller can not longer fire-and-forget. Maybe it makes sense to start migrating methods to return proper errors.

Reply:
you can still return the `error` here (and any new functions that you deem necessary) and just have the caller ignore it, that way when the caller is ready to transition to start checking, it can. Just one less thing to worry about at that point.

Reply:
@ali-farooq0 How would the caller properly check the error? Something like 
```golang
go func() {
	err := CleanUpLabels(...)
	if err != nil {
		Log.Stuff(...)
	}
}()
```

I'm unsure how much I would like that pattern.

Reply:
Fair point. I only recommended returning the `error` if the idea was to handle all the errors at the top level at some point (and migrate other functions to do the same). But doing that would mean that you'd have to do what you highlighted above, which is not the cleanest. Usually you return an error when you want the caller to stop the operation when it receives the error, but in this case it'll be handled through the logs it seems.